pool.js

```diff
@@ -21,14 +22,14 @@ var GO, // GO is global object, for passing to a REPL or finding in a core dump
 //    max_retries: number (default = 5)
 //    agent_options: {} an object for passing options directly to the HTTP(S) Agent
 // }
-function Pool(http, endpoints, options) {
-    if (!http || !http.request || !http.Agent) {
+function Pool(protocol, endpoints, options) {
+    if (!protocol || !protocol.request || !protocol.Agent) {
         throw new Error("invalid http module");
     }
     if (! Array.isArray(endpoints)) {
         throw new Error("endpoints must be an array");
     }
-    this.http = http;
+    this.protocol = protocol;
```

Comment:
nit: property name `http` is more clearly a module to me than `protocol` which sounds like it is going to be a string `"http"` or `"https"`